graphistry/pygraphistry.py

```diff
@@ -1824,6 +1824,23 @@ def tigergraph(
             protocol, server, web_port, api_port, db, user, pwd, verbose
         )
 
+
+    @staticmethod
+    def spannergraph(project_id, instance_id, database_id):
+        """
+        Create a new PlotterBase instance with SpannerGraph configured.
+
+        Args:
+            project_id (str): Google Cloud project ID.
+            instance_id (str): Spanner instance ID.
+            database_id (str): Spanner database ID.
+
+        Returns:
+            PlotterBase: A PlotterBase instance configured with SpannerGraph.
+        """
+        return Plotter().spannergraph(project_id, instance_id, database_id)
```

Comment:
re:interface --


## Connection

i wonder if we want something similar to the bolt connectors, where we can do streamlined auth + passing in a client


```python
CONN_CFG = { ... }
g1 = graphistry.spanner_connect(**CONN_CFG)
g2 = g1.spanner_query("...")
```

+

```python
native_client = google.spanner.client(...)
g1 = graphistry.spanner_connect(native_client)
g2 = g1.spanner_query("....")
```

## Type queries

Separately, a pattern I'm finding to be type-friendly with remote-mode gfql has been separating return-shape-typed methods:

```python
g2 = g1.spanner_g("get a graph")
df = g1.spanner_df("get a table"")
```

Reply:
Also, in the neo4j & gremlin cases, when building real examples:

* hydration: I found scenarios like we'd get either node IDs or edge IDs and would want to then hydrate them against the db
* stored procedures: does GSQL have some notion of passing inputs
* reads vs creates vs writes: any notions here? a good flow would be an example of going through uploading some data, querying it, doing some local enrichment, and then writing back the updates

Part of this makes me want to have some bigger examples to motivate helper functions on top. I suspect `query_g()` and `query_df()` can be good starts, but we need these examples to make more realistic

(The lack of bulk reads/writes such as via arrow is a bit puzzling, but that can wait?)